control-plane-operator/controllers/hostedcontrolplane/konnectivity/reconcile.go

```diff
@@ -236,6 +236,7 @@ func reconcileWorkerAgentDaemonSet(daemonset *appsv1.DaemonSet, deploymentConfig
 				SecurityContext: &corev1.PodSecurityContext{
 					RunAsUser: pointer.Int64Ptr(1000),
 				},
+				HostNetwork: true,
```

Comment:
necessary from what I saw for HTTPS traffic to work through the proxy. (I saw https traffic failing otherwise).

Reply:
so I did this @mihalysz22 because without it HTTPS traffic from the openshift-apiserver -> the openshift-registry was not working (a critical communication path). 

I also noticed that is the recommended deployment upstream so I'm wondering if not running it as host network is potentially causing some instablity

Reply:
this might be the source of some instability if it is determined the agent dameonsets need to run host network